.github/workflows/benchmark-releases.yaml

```diff
@@ -0,0 +1,47 @@
+---
+name: Benchmarks nightly between last release and current main on AMD64
+permissions: read-all
+on:
+  schedule:
+    - cron: '10 5 * * *' # runs every day at 05:10 UTC
+  # workflow_dispatch enables manual testing of this job by maintainers
+  workflow_dispatch:
+jobs:
+  benchmark-nightly:
+    runs-on: ubuntu-latest-8-cores
+    steps:
+    - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+      with:
+        fetch-depth: 0
+    - id: goversion
+      run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+    - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+      with:
+        go-version: ${{ steps.goversion.outputs.goversion }}
+    - name: Run Benchmarks
+      run: |
+        BENCHSTAT_OUTPUT_FILE=result.txt make test-benchmark-compare REF=release-1.3
```

Comment:
When we cut branch release-1.4, then we should manually update to `REF=release-1.4` on the main branch?

I suggest to add a `doc/cut-branch-notes.md` to document all such changes when cutting branch. It can be addressed in a separate PR.

Reply:
I can update it, I'll also refactor the workflow real quick to be shared with parameters